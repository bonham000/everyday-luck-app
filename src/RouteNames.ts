export enum ROUTE_NAMES {
  HOME = "HOME",
  VIEW_NOTE = "VIEW_NOTE",
  CREATE_NOTE = "CREATE_NOTE",
}
