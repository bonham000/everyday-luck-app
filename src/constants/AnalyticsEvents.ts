/** ========================================================================
 * App events for tracking
 * =========================================================================
 */

enum EVENTS {
  COMPLETED_LESSON = "COMPLETED_LESSON",
  SET_APP_DIFFICULTY = "SET_APP_DIFFICULTY",
  SET_APP_LANGUAGE = "SET_APP_LANGUAGE",
  TOGGLE_AUDIO_SETTING = "TOGGLE_AUDIO_SETTING",
  TOGGLE_AUTO_PROCEED_QUIZ = "TOGGLE_AUTO_PROCEED_QUIZ",
}

/** ========================================================================
 * Export
 * =========================================================================
 */

export default EVENTS;
