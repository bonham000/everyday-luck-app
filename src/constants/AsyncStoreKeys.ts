/** ========================================================================
 * Async Store Keys
 * =========================================================================
 */
export const ASYNC_STORE_KEYS = {
  USER_KEY: "USER_KEY",
  WORD_STUDY_LIST: "WORD_STUDY_LIST",
  CUSTOM_WORD_STUDY_LIST: "CUSTOM_WORD_STUDY_LIST",
  DOCUMENTS_AGREEMENT: "DOCUMENTS_AGREEMENT",
  OFFLINE_UPDATES_FLAG: "OFFLINE_UPDATES_FLAG",
};
