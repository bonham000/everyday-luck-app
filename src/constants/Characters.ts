export const CHINESE_NUMBER_MAP: ReadonlyArray<any> = [
  "另",
  "一",
  "二",
  "三",
  "四",
  "五",
  "六",
  "七",
  "八",
  "九",
  "十",
  "十一",
  "十二",
  "十三",
  "十四",
  "十五",
  "十六",
  "十七",
  "十八",
  "十九",
  "二十",
  "二十一",
];
