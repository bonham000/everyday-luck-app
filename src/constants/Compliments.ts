/** ========================================================================
 * Adjective options for when users pass/fail questions.
 * =========================================================================
 */
export const COMPLIMENTS: ReadonlyArray<string> = [
  "Amazing",
  "Awesome",
  "Excellent",
  "Fabulous",
  "Fantastic",
  "Incredible",
  "Outstanding",
  "Perfect",
  "Remarkable",
  "Spectacular",
  "Splendid",
  "Stellar",
  "Stupendous",
  "Super",
  "Unbelievable",
];

export const ENCOURAGEMENTS: ReadonlyArray<string> = [
  "No good",
  "Nope",
  "Too bad",
  "So close",
  "Almost",
  "One more time",
  "Nearly",
  "Try again",
];
