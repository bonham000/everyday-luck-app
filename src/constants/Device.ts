import { Dimensions } from "react-native";

/** ========================================================================
 * Dimension adjustments for smaller screens
 * =========================================================================
 */

const HEIGHT = Dimensions.get("window").height;

const SMALL_DEVICE = HEIGHT < 725;

/** ========================================================================
 * Export
 * =========================================================================
 */

export { SMALL_DEVICE };
