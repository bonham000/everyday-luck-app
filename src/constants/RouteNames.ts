/** ========================================================================
 * Route names
 * =========================================================================
 */
export enum ROUTE_NAMES {
  APP = "APP",
  HOME = "HOME",
  LIST_SUMMARY = "LIST_SUMMARY",
  LESSON_SUMMARY = "LESSON_SUMMARY",
  QUIZ = "QUIZ",
  DAILY_CHALLENGE = "DAILY_CHALLENGE",
  HSK_TEST_OUT = "HSK_TEST_OUT",
  MULTIPLE_CHOICE_MANDARIN = "MULTIPLE_CHOICE_MANDARIN",
  MULTIPLE_CHOICE_ENGLISH = "MULTIPLE_CHOICE_ENGLISH",
  MULTIPLE_CHOICE_VOICE = "MULTIPLE_CHOICE_VOICE",
  FLASHCARDS = "FLASHCARDS",
  VIEW_ALL = "VIEW_ALL",
  TRANSLATION = "TRANSLATION",
  WELCOME = "WELCOME",
  INTRO = "INTRO",
  ABOUT = "ABOUT",
  CONTACT = "CONTACT",
  SETTINGS = "SETTINGS",
}
