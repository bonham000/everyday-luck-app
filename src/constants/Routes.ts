/** ========================================================================
 * Route names
 * =========================================================================
 */
export enum ROUTE_NAMES {
  HOME = "HOME",
  LESSON_SUMMARY = "LESSON_SUMMARY",
  QUIZ = "QUIZ",
  FLASHCARDS = "FLASHCARDS",
  VIEW_ALL = "VIEW_ALL",
  ABOUT = "ABOUT",
}
