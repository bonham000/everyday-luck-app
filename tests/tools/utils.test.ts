import { ROUTE_NAMES } from "@src/constants/RouteNames";
import {
  APP_DIFFICULTY_SETTING,
  APP_LANGUAGE_SETTING,
} from "@src/providers/GlobalStateContext";
import {
  capitalize,
  convertAppDifficultyToLessonSize,
  formatUserLanguageSetting,
  getAlternateLanguageSetting,
  getAudioFileUrl,
  getListScoreKeyFromIndex,
  isLessonComplete,
  mapWordsForList,
  resetNavigation,
} from "@src/tools/utils";
import { DEFAULT_SCORE_STATE, MOCK_WORD } from "@tests/data";

describe("utils", () => {
  test("resetNavigation", () => {
    const result = resetNavigation(ROUTE_NAMES.HOME);
    expect(result).toMatchSnapshot();
  });

  test("mapWordsForList", () => {
    const result = mapWordsForList(MOCK_WORD);
    expect(result).toMatchSnapshot();
  });

  test("getListScoreKeyFromIndex", () => {
    const results: ReadonlyArray<any> = [
      getListScoreKeyFromIndex(0),
      getListScoreKeyFromIndex(1),
      getListScoreKeyFromIndex(2),
      getListScoreKeyFromIndex(3),
      getListScoreKeyFromIndex(4),
    ];

    for (const result of results) {
      expect(result).toMatchSnapshot();
    }
  });

  test("isLessonComplete", () => {
    expect(isLessonComplete(DEFAULT_SCORE_STATE)).toBeFalsy();

    expect({
      mc_english: true,
      mc_mandarin: true,
      quiz_text: true,
      mandarin_pronunciation: true,
    }).toBeTruthy();
  });

  test("formatUserLanguageSetting", () => {
    expect(
      formatUserLanguageSetting(APP_LANGUAGE_SETTING.SIMPLIFIED),
    ).toMatchInlineSnapshot(`"Simplified Chinese"`);
    expect(
      formatUserLanguageSetting(APP_LANGUAGE_SETTING.TRADITIONAL),
    ).toMatchInlineSnapshot(`"Traditional Chinese"`);
  });

  test("getAlternateLanguageSetting", () => {
    expect(getAlternateLanguageSetting(APP_LANGUAGE_SETTING.TRADITIONAL)).toBe(
      APP_LANGUAGE_SETTING.SIMPLIFIED,
    );
    expect(getAlternateLanguageSetting(APP_LANGUAGE_SETTING.SIMPLIFIED)).toBe(
      APP_LANGUAGE_SETTING.TRADITIONAL,
    );
  });

  test("getAudioFileUrl", () => {
    const formatUrlResult = (url: string) => url.slice(url.indexOf("static"));
    let result = getAudioFileUrl("我-1");
    expect(formatUrlResult(result)).toMatchInlineSnapshot(
      `"static/%E6%88%91-1.mp3"`,
    );

    result = getAudioFileUrl("蛋糕-5");
    expect(formatUrlResult(result)).toMatchInlineSnapshot(
      `"static/%E8%9B%8B%E7%B3%95-5.mp3"`,
    );

    result = getAudioFileUrl("對不起-3");
    expect(formatUrlResult(result)).toMatchInlineSnapshot(
      `"static/%E5%B0%8D%E4%B8%8D%E8%B5%B7-3.mp3"`,
    );
  });

  test("convertAppDifficultyToLessonSize", () => {
    expect(
      convertAppDifficultyToLessonSize(APP_DIFFICULTY_SETTING.EASY),
    ).toMatchInlineSnapshot(`10`);
    expect(
      convertAppDifficultyToLessonSize(APP_DIFFICULTY_SETTING.MEDIUM),
    ).toMatchInlineSnapshot(`20`);
    expect(
      convertAppDifficultyToLessonSize(APP_DIFFICULTY_SETTING.HARD),
    ).toMatchInlineSnapshot(`30`);
  });

  test("capitalize", () => {
    expect(capitalize("hello hello hello")).toBe("Hello hello hello");
    expect(capitalize("Hello HELLO HELLO")).toBe("Hello hello hello");
    expect(capitalize("HELLO hello hello")).toBe("Hello hello hello");
    expect(capitalize("hELLO hello hello")).toBe("Hello hello hello");
  });
});
