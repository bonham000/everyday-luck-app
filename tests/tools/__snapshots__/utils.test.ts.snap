// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils getListScoreKeyFromIndex 1`] = `"list_02_score"`;

exports[`utils getListScoreKeyFromIndex 2`] = `"list_03_score"`;

exports[`utils getListScoreKeyFromIndex 3`] = `"list_04_score"`;

exports[`utils getListScoreKeyFromIndex 4`] = `"list_05_score"`;

exports[`utils getListScoreKeyFromIndex 5`] = `"list_06_score"`;

exports[`utils mapWordsForList 1`] = `
Object {
  "english": "aunt",
  "english_alternate_choices": Array [
    "",
  ],
  "key": "阿姨",
  "pinyin": "āyí",
  "simplified": "阿姨",
  "traditional": "阿姨",
}
`;
